### PR TITLE
Christodouloumass

### PR DESCRIPTION
**Pull request by trami18**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: docs/References.bib [hunk 1]
```diff
@@ -403,3 +403,20 @@ @article{Yee1999
...
+      volume         = "33",
+      year           = "2016",
+      number         = "24",
+      pages          = "244002",
+      doi            = "10.1088/0264-9381/33/24/244002",
+      eprint         = "1607.05377",
+      archivePrefix  = "arXiv",
+      primaryClass   = "gr-qc",
+      SLACcitation   = "%%CITATION = ARXIV:1607.05377;%%"
+      url            = "https://arxiv.org/pdf/1607.05377.pdf"
```
Comment (geoffrey4444) [inline docs/References.bib]: url -> "https://doi.org/10.1088/0264-9381/33/24/244002"

File: docs/References.bib [hunk 2]
```diff
@@ -403,3 +403,20 @@ @article{Yee1999
 # When editing this file, please follow the guidelines in
 # https://spectre-code.org/writing_good_dox.html#writing_dox_citations.
 
+}
+@article{Lovelace:2016uwp,
```
Comment (geoffrey4444) [inline docs/References.bib]: No colon

Comment (geoffrey4444) [inline docs/References.bib]: Please put this reference in alphabetical order with the other entries by the key

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 3]
```diff
@@ -303,4 +303,17 @@ std::array<double, 3> spin_vector(double spin_magnitude,
  * \f$M_\mathrm{irr}=\sqrt{\frac{A}{16\pi}}\f$.
  */
 double irreducible_mass(double area) noexcept;
+
+/*!
+ * \ingroup SurfacesGroup
+ * \brief Christodulou Mass of a 2D `Strahlkorper`.
+ *
+ * \details See Eq. (1) of e.g \cite Lovelace:2016uwp.
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: @trami18 says no colon here

Comment (geoffrey4444) [reply src/ApparentHorizons/StrahlkorperGr.hpp]: "See e.g. Eq. (1) of \cite Lovelace:2016uwp"

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 4]
```diff
@@ -303,4 +303,17 @@ std::array<double, 3> spin_vector(double spin_magnitude,
  * \f$M_\mathrm{irr}=\sqrt{\frac{A}{16\pi}}\f$.
  */
 double irreducible_mass(double area) noexcept;
+
+/*!
+ * \ingroup SurfacesGroup
+ * \brief Christodulou Mass of a 2D `Strahlkorper`.
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: Christodoulou here and throughout the PR

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 5]
```diff
@@ -303,4 +303,17 @@ std::array<double, 3> spin_vector(double spin_magnitude,
...
+ *
+ * \details See Eq. (1) of e.g \cite Lovelace:2016uwp.
+ * This function computes the Christodulou mass from the spin angular momentum
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: spin angular momentum -> dimensionless spin angular momentum

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 6]
```diff
@@ -303,4 +303,17 @@ std::array<double, 3> spin_vector(double spin_magnitude,
...
+ *`$\chi$` and the irreducible mass `$M_{irr}$` of a black hole horizon.
+ * Specifically, computes
+ * \f$M_{Chris}= \sqrt{\frac{M_{irr}^2+\chi^2}{4M_{irr}^2}}
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: M^2 = Mirr^2 + \chi^2 M^4 / 4 Mirr^2, solve for M, make sure this function gives M. \chi = S / M^2, so you have to solve for S and plug into Eq. (1) then solve for M.

File: src/ApparentHorizons/StrahlkorperGr.cpp [hunk 7]
```diff
@@ -660,6 +660,13 @@ double irreducible_mass(const double area) noexcept {
          "The area of the horizon must be greater than zero but is " << area);
   return sqrt(area / (16.0 * M_PI));
 }
+
+}
+double square(double a) {
+  return a*a;
+}
+double christodoulou_mass(const double  spin_magnitude) noexcept {
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: spin_magnitude -> dimensionless_spin_magnitude, const double irreducible_mass

File: src/ApparentHorizons/StrahlkorperGr.hpp [hunk 8]
```diff
@@ -303,4 +303,17 @@ std::array<double, 3> spin_vector(double spin_magnitude,
...
+ */
+double christodoulou_mass(double dimensionless_spin_magnitude,
+                          irreducible_mass) noexcept;
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.hpp]: double irreducible_mass

File: src/ApparentHorizons/StrahlkorperGr.cpp [hunk 9]
```diff
@@ -660,6 +660,13 @@ double irreducible_mass(const double area) noexcept {
...
+  return sqrt(square(irreducible_mass)+ square(dimensionless_spin_magnitude)/ (4.0 * square(irreducible_mass)));
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: Always run git-clang-format -f on your code before committing. In this case, add a space after the line to change it, so git-clang-format -f sees that this is a modified line.

File: src/ApparentHorizons/StrahlkorperGr.cpp [hunk 10]
```diff
@@ -660,6 +660,13 @@ double irreducible_mass(const double area) noexcept {
          "The area of the horizon must be greater than zero but is " << area);
   return sqrt(area / (16.0 * M_PI));
 }
+
+}
+double square(double a) {
```
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: Don't need square()
Just need to `#include Utilities/ConstantExpressions.hpp`

File: tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp [hunk 11]
```diff
@@ -249,8 +249,8 @@ void test_area_element(const Solution& solution, const double& surface_radius,
 
 template <typename Solution, typename Fr>
 void test_area(const Solution& solution, const Strahlkorper<Fr>& strahlkorper,
-               const double expected,
-               const double expected_irreducible_mass) noexcept {
+               const double expected, const double expected_irreducible_mass,
+               const double expected_christodouloumass) noexcept {
```
Comment (geoffrey4444) [inline tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp]: expected_christodoulou_mass (add underscore)

File: tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp [hunk 12]
```diff
@@ -283,6 +283,10 @@ void test_area(const Solution& solution, const Strahlkorper<Fr>& strahlkorper,
 
   const double irreducible_mass = StrahlkorperGr::irreducible_mass(area);
   CHECK(irreducible_mass == approx(expected_irreducible_mass));
+
+  const double christodoulou_mass = StrahlkorperGr::christodoulou_mass(
+      dimensionless_spin_magnitude, irreducible_mass);
```
Comment (geoffrey4444) [inline tests/Unit/ApparentHorizons/Test_StrahlkorperGr.cpp]: pass dimensionless_spin_magnitude to test_area(), or pass spin and take magnitude here

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
Current status:

- [ ] @geoffrey4444 needs to look at the fix ups
- [ ] @geoffrey4444 will help with the rebase and squash when happy with the fix ups

**Comment by geoffrey4444** [comment]
Open

**Pull request closed without merging**